spacy/strings.pyx

```diff
@@ -14,6 +14,13 @@ from .symbols import NAMES as SYMBOLS_BY_INT
 from .errors import Errors
 from . import util
 
+# Not particularly elegant, but this is faster than `isinstance(key, numbers.Integral)`
```

Comment:
Is this faster under some conditions, like if you mostly don't hit the except block?

Reply:
If I recall correctly, this version was always faster than the `isinstance` check (the former barely even showed up in the profile). I think it might have to do with how the `Integral` abstract base class is implemented.

Reply:
Ah OK, sounds good then.